src/audio/module_adapter/module/waves/waves.c

```diff
@@ -456,6 +458,50 @@ static int waves_effect_revision(struct processing_module *mod)
 	return 0;
 }
 
+/* cache config blob*/
+static int waves_effect_cache_config_blob(struct processing_module *mod, void *data, uint32_t size)
+{
+	struct comp_dev *dev = mod->dev;
+	struct module_data *codec = &mod->priv;
+	struct waves_codec_data *waves_codec = codec->private;
+	int ret = 0;
+
+	comp_dbg(dev, "waves_effect_cache_config_blob() start");
+
+	/* release old cached config blob*/
+	if (waves_codec->config_blob && size != waves_codec->config_blob_size) {
+		comp_info(dev, "waves_effect_cache_config_blob() release blob");
+		module_free_memory(mod, waves_codec->config_blob);
+		waves_codec->config_blob = NULL;
+		waves_codec->config_blob_size = 0;
+	}
+
+	if (!waves_codec->config_blob) {
+		waves_codec->config_blob = module_allocate_memory(mod, size, 16);
+		if (!waves_codec->config_blob) {
+			comp_err(dev,
+				"waves_effect_cache_config_blob() failed to allocate %d bytes for config blob",
+				size);
+			ret = -ENOMEM;
```

Comment:
I'd just do `return -ENOMEM` here (and below) and also avoid a `done` debug print in an error case, remove the `ret` variable